File: roguedoom_classic/Cell.java
import java.util.*;

class Cell
{
	public static final int NORMALROOM = 0;	
	public static final int REDBASE = 1;
	public static final int BLUEBASE = 2;
	public static final int REDALTAR = 3;
	public static final int BLUEALTAR = 4;
	public static final int ELEVATOR= 5;
	public static final int ELEVATORROOM= 6;
	public static final int REDSOURCE= 7;
	public static final int BLUESOURCE= 8;
	
	public boolean EastBound,SouthBound;
	public int mark;
	
	public int m_row,m_column;
	public boolean m_upstairs;
	public Vector m_entities;
	/////para calculos de visibilidad
	double leftTangent=0, rightTangent=0;	
	public boolean mark2;	
	
	/*
	0 top
	1 bottom
	2 left
	3 right
	*/
	
	
	public Cell()
	{
		mark= Cell.NORMALROOM;
		mark2 = false;
		EastBound=false;
		SouthBound=false;		
		m_entities = new Vector();
		
	}	
	
	
	
	
	
}

File: src/Cell.js
export class Cell {
    // Cell types
    static NORMALROOM = 0;
    static REDBASE = 1;
    static BLUEBASE = 2;
    static ELEVATOR = 3; // The moving platform cell
    static ELEVATORROOM = 4; // Cells surrounding the elevator platform
    static OUTSIDE = 5;
    static REDALTAR = 6;
    static BLUEALTAR = 7;
    static REDSOURCE = 8;
    static BLUESOURCE = 9;
    // Add more as needed

    constructor(row, column, upstairs) {
        this.EastBound = false; // true if there is a wall to the east
        this.SouthBound = false; // true if there is a wall to the south

        this.mark = Cell.NORMALROOM; // type of cell
        this.m_row = row;
        this.m_column = column;
        this.m_upstairs = upstairs; // true if the cell is on the upper floor

        this.m_entities = []; // list of 3D objects in this cell

        // For visibility determination
        this.leftTangent = 0.0;
        this.rightTangent = 0.0;
        this.mark2 = false; // used by visibility algorithms
    }
}
